pgjdbc/src/main/java/org/postgresql/jdbc/EscapedFunctions.java

```diff
@@ -769,24 +730,33 @@ public static String sqluser(List<?> parsedArgs) throws SQLException {
 
   private static String singleArgumentFunctionCall(String call, String functionName,
       List<?> parsedArgs) throws PSQLException {
-    StringBuilder buf = new StringBuilder();
-    buf.append(call);
     if (parsedArgs.size() != 1) {
       throw new PSQLException(GT.tr("{0} function takes one and only one argument.", functionName),
           PSQLState.SYNTAX_ERROR);
     }
-    buf.append(parsedArgs.get(0));
+    StringBuilder buf = new StringBuilder(call.length() + 30); //heuristics !!
```

Comment:
I wonder if `StringBuilder` could be received as a argument like `newSql` from https://github.com/pgjdbc/pgjdbc/blob/38356e6889613a65fc48a455495f18dbb3565731/pgjdbc/src/main/java/org/postgresql/core/Parser.java#L1194

Reply:
Left as an exercise for the reader !
Another patch should also change the functions signature to take a List<CharSequence> instead of List<?>

Please merge this PR and open an issue with your ideas for future improvements.